crates/wasi-nn/src/backend/winml.rs

```diff
@@ -226,3 +205,180 @@ fn convert_i64(i: i64) -> Result<u32, BackendError> {
         anyhow::anyhow!("unable to convert dimension to u32: {d}").into()
     })
 }
+
+// Convert from wasi-nn tensor to WinML tensor.
+fn to_inspectable(tensor: &Tensor, shape: IVectorView<i64>) -> Result<IInspectable, Error> {
+    match tensor.ty {
+        crate::wit::types::TensorType::Fp16 => unsafe {
+            let data = std::slice::from_raw_parts(
+                tensor.data.as_ptr() as *const f32,
+                tensor.data.len() / size_of::<f32>(),
+            );
+            TensorFloat16Bit::CreateFromArray(&shape, data)?.cast::<IInspectable>()
+        },
+        crate::wit::types::TensorType::Fp32 => unsafe {
+            let data = std::slice::from_raw_parts(
+                tensor.data.as_ptr() as *const f32,
+                tensor.data.len() / size_of::<f32>(),
+            );
+            TensorFloat::CreateFromArray(&shape, data)?.cast::<IInspectable>()
+        },
+        crate::wit::types::TensorType::I64 => unsafe {
+            let data = std::slice::from_raw_parts(
+                tensor.data.as_ptr() as *const i64,
+                tensor.data.len() / size_of::<i64>(),
+            );
+            TensorInt64Bit::CreateFromArray(&shape, data)?.cast::<IInspectable>()
+        },
+        _ => unimplemented!(),
+    }
+}
+
+// Convert from WinML tensor to wasi-nn tensor.
+fn to_tensor(inspectable: IInspectable, tensor_kind: TensorKind) -> Result<Tensor, BackendError> {
+    let tensor = match tensor_kind {
+        TensorKind::Float16 => {
+            let output_tensor = inspectable.cast::<TensorFloat16Bit>()?;
+            let dimensions = dimensions_as_u32(&output_tensor.Shape()?)?;
```

Comment:
We can do this once at the top of the function.

Reply:
`output_tensor`'s type is unknown at the top of the function.

Reply:
How about

```rust
let itensor = inspectable.cast<ITensor>()?;
let dimensions = dimensions_as_u32(&itensor.Shape()?)?;
```

or something like that?

Reply:
It works, but `GetAsVectorView` is a method of `TensorFloat16Bit`. Then we'll need to cast `itensor` again to specific tensor type. The change above makes code clean but will it be a performance issue to cast twice?